PyInstaller/loader/pyiboot01_bootstrap.py

```diff
@@ -46,13 +46,17 @@
 
 # Some packages behaves differently when running inside virtual environment.
 # E.g. IPython tries to append path VIRTUAL_ENV to sys.path.
-# For the frozen app we want to prevent this behavior.
-VIRTENV = 'VIRTUAL_ENV'
-if VIRTENV in os.environ:
-    # On some platforms (e.g. AIX) 'os.unsetenv()' is not available and then
-    # deleting the var from os.environ does not delete it from the environment.
-    os.environ[VIRTENV] = ''
-    del os.environ[VIRTENV]
+# For the frozen app we want to prevent this behavior,
+# but we still preserve original values in case they are required by the application.
+PRESERVED_ENVS = '{{ pyiboot01_preserved_envs }}'
+PRESERVED_ENV_PREFIX = 'PYINSTALLER_PRESERVED_'
+for env in PRESERVED_ENVS:
+    if env in os.environ:
+        # On some platforms (e.g. AIX) 'os.unsetenv()' is not available and then
+        # deleting the var from os.environ does not delete it from the environment.
+        os.environ[PRESERVED_ENV_PREFIX + env] = os.environ[env]
+        os.environ[env] = ''
+        del os.environ[env]
```

Comment:
Please add `del PRESERVED_ENVS, PRESERVED_ENV_PREFIX` after the loop

Reply:
Fixed, I've considered that before, but noticing other temporary vars are not cleaned up let me gave up that idea.